### PR TITLE
CollegiateSubredditsTable Component

### LINKED ISSUE #7: Table component for CollegiateSubreddits
Table component for CollegiateSubreddits

Using the `UCSBDatesTable` component as a model, create a similar component for CollegiateSubreddits.

# ACs

- [x] The table has columns for database fields (i.e. the attributes of the corresponding `@Entity` class
- [x] There are fixtures under `/frontend/src/fixtures` for this data type, similar to the other fixtures files.
- [x] There is a storybook entry similar to the one for `UCSBDatesTable`
- [x] 100% coverage with `npm run coverage`
- [x] 100% coverage with `npx stryker run`

### PR DESCRIPTION
**Pull request by isaacchin**
# Overview
Created CollegiateSubredditsTable component with columns for database fields 
Added fixtures for CollegiateSubredditsTable 
Added storybook entry for CollegiateSubredditsTable 
Added tests for CollegiateSubredditsTable with 100% scores with npm run coverage and npx stryker run

# Issues Addressed
#7 

# Details
Test coverage: image
Mutation testing: image

## Reminders (delete this section before merging or as code is merged)
- [x] Explicitly link to this pull request each issue that it closes
- [x] Assign one and ideally two reviewers to review this pull request. 
- [x] Reviewers should not just approve with "Looks good!" Instead, reviewers should carefully go through the code and use GitHub's reviewer interface to ask questions, make comments, and suggest changes.

### CODE REVIEW

**Comment by mglush** [comment]
resolved conflicts in the collegiate subreddit fixtures file.

File: frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      });
+  
+      expectedFields.forEach((field) => {
+        const header = getByTestId(`${testId}-cell-row-0-col-${field}`);
+        expect(header).toBeInTheDocument();
+      });
+  
+      expect(getByTestId(`${testId}-cell-row-0-col-id`)).toHaveTextContent("1");
+      expect(getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("2");
+
```
Comment (mglush) [inline frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js]: Why do you have 3 collegiate subreddits in your fixtures, load up 3 collegiate subreddits into the table, and then only check 1 and 2???

Add this on line 93:
expect(getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("3");

File: frontend/src/main/components/CollegiateSubreddits/CollegiateSubredditsTable.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    // Stryker disable next-line ArrayDeclaration : [columns] is a performance optimization
+    const memoizedColumns = React.useMemo(() => columns, [columns]);
+    const memoizedDates = React.useMemo(() => subreddits, [subreddits]);
+
+    return <OurTable
+        data={memoizedDates}
+        columns={memoizedColumns}
+        testid={"CollegiateSubredditsTable"}
+    />;
+};
```
Comment (mglush) [inline frontend/src/main/components/CollegiateSubreddits/CollegiateSubredditsTable.js:59]: all good here.

File: frontend/src/main/utils/CollegiateSubredditsUtils.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export function cellToAxiosParamsDelete(cell) {
+    return {
+        url: "/api/collegiatesubreddits",
+        method: "DELETE",
+        params: {
+            id: cell.row.values.id
+        }
+    }
+}
+
```
Comment (mglush) [inline frontend/src/main/utils/CollegiateSubredditsUtils.js:18]: all good here.

File: frontend/src/stories/components/CollegiateSubreddits/CollegiateSubredditsTable.stories.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    subjects: []
+};
+
+export const ThreeSubjects = Template.bind({});
+
+ThreeSubjects.args = {
+    subjects: collegiateSubredditsFixtures.threeSubreddits
+};
+
+
```
Comment (mglush) [inline frontend/src/stories/components/CollegiateSubreddits/CollegiateSubredditsTable.stories.js:29]: this file looks all good.

File: frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  
+      await waitFor(() => expect(mockedMutate).toHaveBeenCalledTimes(1));
+  
+  
+    });
+  
+  
+  
+  });
+
```
Comment (mglush) [inline frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js:157]: file requires some changes.

**Review by mglush — CHANGES_REQUESTED** [review]
Need to change the test file to check the id of all 3 entries you loaded up, not just 2.

**Review by stychoi — COMMENTED** [review]
Other than the thing Glush pointed out, LGTM

File: frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js (hunk 1 above)
Comment (isaacchin) [reply frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js]: Good catch, I pushed the correction to the test now

**Review by mglush — APPROVED** [review]

**Pull request merged**